src/main/kotlin/com/cognifide/gradle/aem/base/vlt/VltFilter.kt

```diff
@@ -22,12 +25,37 @@ class VltFilter(val file: File, private val temporary: Boolean = false) : Closea
             return VltFilter(file, true)
         }
 
+        fun rootElement(xml: String): Element {
+            return Jsoup.parse(xml, "", Parser.xmlParser()).select("filter[root]").first()
+        }
+
+        fun rootElementForPath(path: String): Element {
+            return rootElement("<filter root=\"$path\"/>")
+        }
+
     }
 
+    val rootElements: Set<Element>
+        get() {
+            val xml = file.bufferedReader().use { it.readText() }
+            val doc = Jsoup.parse(xml, "", Parser.xmlParser())
+
+            return doc.select("filter[root]").toSet()
+        }
+
+    val rootPaths: Set<String>
```

Comment:
by reading this, we are cleaning only files that are check out. Previously always all files was cleaned so that `aemSync` now will work faster and will touch only changed files.